src/main/java/io/jenkins/plugins/pipelinegraphview/utils/AbstractPipelineViewAction.java

```diff
@@ -57,6 +113,37 @@ public boolean isParameterized() {
         return property != null && !property.getParameterDefinitions().isEmpty();
     }
 
+
+     /**
+     * Handles the rebuild request and redirects to parameterized
+     * and non parameterized build when needed.
+     */
+    @RequirePOST
+    @JavaScriptMethod
+    public void doRebuildjob()  {
+        if (run != null) {
+            run.checkPermission(Item.BUILD);
```

Comment:
This is how the Pipeline plugin does replay:

https://github.com/jenkinsci/workflow-cps-plugin/blob/5b32b1a2e5b83a23354bbc147037cbdbc9d9061f/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java#L272-L298

(we don't want the script modification bit)

Need to dig out how blueocean does it as well

Reply:
@timja I will study it, I have implemented using some idea from rebuild plugin and change it. Now, I will take a look there

Reply:
@timja Is somenthing that is the wrap in some class that I can use from here?

Reply:
I don't understand what you mean, the code won't be re-usable as is but relevant bits may be copied

Reply:
@timja I have update the proposal. Let me know if it's a direction that you like more

Reply:
Thanks, I wonder if you could just do this? https://github.com/jenkinsci/blueocean-plugin/blob/2b1ec08517580d4620222183d9d1b3daadd9efa2/blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java#L111C51-L116

Reply:
@timja I need to check with that class I can make the code smaller I think

Reply:
@timja Thank you, I keep down the code now. I have some problem on some path still. For instance do you like the javascript or you want to go to strampler2 interface?